From 6a5baa38289acbd7d822bfa6e47ed038ba4d5002 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludovic=20Court=C3=A8s?= <ludo@gnu.org>
Date: Mon, 6 Feb 2023 14:38:18 +0100
Subject: [PATCH] Rename 'scm_pipe2' to 'scm_fibers_pipe2'.

This fixes a build error with Guile 3.0.9:

  extensions/epoll.c:224:1: error: static declaration of 'scm_pipe2' follows non-static declaration

* extensions/epoll.c (scm_pipe2): Rename to...
(scm_fibers_pipe2): ... this.
(init_fibers_epoll): Adjust accordingly.
---
 extensions/epoll.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/extensions/epoll.c b/extensions/epoll.c
index 9b6f2bb..cae279e 100644
--- a/extensions/epoll.c
+++ b/extensions/epoll.c
@@ -1,5 +1,5 @@
 /* Copyright (C) 2016 Andy Wingo <wingo@pobox.com>
- * Copyright (C) 2022 Ludovic Courtès <ludo@gnu.org>
+ * Copyright (C) 2022, 2023 Ludovic Courtès <ludo@gnu.org>
  * Copyright (C) 2022 Aleix Conchillo Flaqué <aconchillo@gmail.com>
  *
  * This library is free software; you can redistribute it and/or
@@ -220,9 +220,11 @@ scm_primitive_epoll_wait (SCM epfd, SCM wakefd, SCM wokefd,
 
 static SCM sym_read_pipe, sym_write_pipe;
 
+/* Note: Guile 3.0.9 introduced an 'scm_pipe2' function, which is marked as
+   'SCM_INTERNAL'.  Use a different name here to avoid a collision.  */
 static SCM
-scm_pipe2 (SCM flags)
-#define FUNC_NAME "scm_pipe2"
+scm_fibers_pipe2 (SCM flags)
+#define FUNC_NAME "scm_fibers_pipe2"
 {
   int c_flags, ret, fd[2];
   SCM read_port, write_port;
@@ -279,7 +281,7 @@ init_fibers_epoll (void)
   scm_c_define ("EPOLL_CTL_MOD", scm_from_int (EPOLL_CTL_MOD));
   scm_c_define ("EPOLL_CTL_DEL", scm_from_int (EPOLL_CTL_DEL));
 
-  scm_c_define_gsubr ("pipe2", 0, 1, 0, scm_pipe2);
+  scm_c_define_gsubr ("pipe2", 0, 1, 0, scm_fibers_pipe2);
   sym_read_pipe = scm_from_latin1_string ("read pipe");
   sym_write_pipe = scm_from_latin1_string ("write pipe");
 
